adapters/eea-direct.js

```diff
@@ -0,0 +1,238 @@
+
+export function fetchData (source, cb) {
+  const metadataRequest = makeMetadataRequest(source);
+  const requestTasks = makeTaskRequests(source);
+  parallel(
+    [metadataRequest, requestTasks],
+    (err, res) => {
+      if (err) {
+        return cb('Error getting data from source', []);
+      }
+      try {
+        formatData(res, source, cb);
+      } catch (e) {
+        cb({message: 'Error parsing the data'}, null);
+      }
+    });
+}
+
+// makes request used to get then format metadata for station coordinates
+const makeMetadataRequest = (source) => {
+  return (cb) => {
+    request.get({
+      url: 'http://discomap.eea.europa.eu/map/fme/metadata/PanEuropean_metadata.csv'
+    }, (err, res, body) => {
+      if (err || res.statusCode !== 200) {
+        return cb('Could not gather current metadata, will generate records without coordinates.', []);
+      }
+      let data;
+      try {
+        data = parse(body, {delimiter: '\t'});
+      } catch (e) {
+        return cb('Could not parse metadata file', []);
+      }
+      getCoordinates(data, source.country, cb);
+    });
+  };
+};
+
+// reduce metadata to list of objects with coordinates for
+const getCoordinates = (metadata, country, callback) => {
+  // filter for only country of interest's records
+  metadata = metadata.filter((record) => {
+    return record[0] === country;
+  });
+  map(metadata, (record, done) => {
+    const station = {
+      stationId: record[5],
+      coordinates: {
+        latitude: parseFloat(record[15]),
+        longitude: parseFloat(record[14])
+      }
+    };
+    done(null, station);
+  }, (err, mappedRecords) => {
+    if (err) {
+      return callback(null, []);
+    }
+    callback(null, uniqBy(mappedRecords, 'stationId'));
+  });
+};
+
+// makes requests to get country's pollutant data.
+const makeTaskRequests = (source) => {
+  const pollutantRequests = acceptableParameters.map((pollutant) => {
+    pollutant = pollutant.toUpperCase();
+    return (done) => {
+      const url = source.url + source.country + '_' + pollutant + '.csv';
+      request.get({
+        url: url
+      }, (err, res, body) => {
+        if (err || res.statusCode !== 200) {
+          return done(null, []);
+        }
+        done(null, parse(body).slice(1, -1));
+      });
+    };
+  });
+  return (done) => {
+    parallel(
+      pollutantRequests,
+      (err, response) => {
+        if (err) {
+          done(null, []);
+        }
+        done(null, [].concat.apply([], response));
+      }
+    );
+  };
+};
+
+// formats data to match openAQ standard
+const formatData = (data, source, cb) => {
+  const coordinates = data[0];
+  const records = data[1];
+  parallel([
+    (done) => {
+      map(records, (record, cb) => {
+        let measurement = {
+          coordinates: makeCoordinates(coordinates, record[11]),
+          parameter: record[5].toLowerCase(),
+          date: makeDate(record[16], record[4]),
+          value: parseInt(record[19]),
```

Comment:
Why are we doing a `parseInt()` here? We're losing information by rounding to an int. Probably want to use `Number()`?